src/Raven.Server/Documents/Revisions/RevisionsStorage.ManuallyDelete.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Server.Documents.TransactionMerger.Commands;
+using Raven.Server.ServerWide.Context;
+using Voron;
+using Voron.Data.Tables;
+
+namespace Raven.Server.Documents.Revisions;
+
+public partial class RevisionsStorage
+{
+    public bool IsAllowedToDeleteRevisionsManually(string collection)
+    {
+        var configuration = GetRevisionsConfiguration(collection);
+        if (configuration == ConflictConfiguration.Default)
+            return false;
+
+        return configuration.AllowDeleteRevisionsManually;
+    }
+
+    public Task DeleteRevisionsByChangeVectorManuallyAsync(List<string> cvs, long maxDeletes, bool shouldThrowIfChangeVectorsNotFound = true)
+    {
+        if (cvs == null || cvs.Count == 0)
+            return Task.CompletedTask;
+
+        if (cvs.Count > maxDeletes)
+            return Task.FromException(new InvalidOperationException($"You are trying to delete more revisions then the limit: {maxDeletes}"));
+
+        return _database.TxMerger.Enqueue(new DeleteRevisionsByChangeVectorManuallyMergedCommand(cvs, shouldThrowIfChangeVectorsNotFound));
+    }
+
+    private void DeleteRevisionsByChangeVectorManuallyInternal(DocumentsOperationContext context, List<string> cvs, bool shouldThrowIfChangeVectorsNotFound)
+    {
+        var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+
+        var table = new Table(context.DocumentDatabase.DocumentsStorage.RevisionsStorage.RevisionsSchema, context.Transaction.InnerTransaction);
+
+        var writeTables = new Dictionary<string, Table>();
+
+        foreach (var cv in cvs)
+        {
+            if (string.IsNullOrEmpty(cv))
+                throw new ArgumentException("Change Vector is null or empty");
+
+            Document revision;
+            using (Slice.From(context.Allocator, cv, out var cvSlice))
+            {
+                if (table.ReadByKey(cvSlice, out TableValueReader tvr) == false)
+                {
+                    if (shouldThrowIfChangeVectorsNotFound == false)
```

Comment:
Why is that even a thing?
We wanted it deleted, it is already is. No harm for us.

Reply:
Done